cocos/core/event/callbacks-invoker.ts

```diff
@@ -41,16 +42,40 @@ class CallbackInfo {
     public once = false;
 
     public set (callback: Function, target?: Object, once?: boolean) {
-        this.callback = callback;
+        this.callback = callback || empty;
         this.target = target;
         this.once = !!once;
     }
+
+    public reset () {
+        this.target = undefined;
+        this.callback = empty;
+        this.once = false;
+    }
+
+    public check () {
+        // Validation
+        if (this.target instanceof CCObject && !this.target.isValid) {
```

Comment:
Only check for CCObject with DESTROYED flag, could be using TO_DESTROY if it's not safe enough. TO_DESTROY flag is set when `destroy()` invoked, DESTROYED flag is set when `deferredDestroy` happened once every frame

Reply:
Can just use `return cc.isValid(this.target);`

> cc.isValid = function (value, strictMode) {
    if (typeof value === 'object') {
        return !!value && !(value._objFlags & (strictMode ? (Destroyed | ToDestroy) : Destroyed));
    }
    else {
        return typeof value !== 'undefined';
    }
};

Reply:
It is not suitable to introduce a coupling between event system and `CCObject`. You could define a special property method for this check:
```ts
class CCObject { __callback_check__() { return this.isValid; } }
class CallbacksInvoker { check() { return this.target.__callback_check__?.() ?? true; } }
```

Reply:
CCObject is a basic class, I don't think it's inappropriate to use it's api here

Reply:
Even it seems basic in engine, the `CallbacksInvoker` should not rely on it.